_posts/2022-05-16-opentelemetry-metrics-for-data-prepper.md

```diff
@@ -0,0 +1,114 @@
+You will be able to augment logs and traces by a careful selection of metrics to enhance your overall observability.
+
+## Solution Architecture
+
+Data Prepper now supports metrics ingestion into OpenSearch using the OpenTelemetry protocol.
+There are many implementations capable of sending metrics using this protocol.
+The OpenTelemetry Collector was chosen as an agent for metrics acquisition in the following diagram.
+It can pull several metrics endpoints or act as a receiver for others.
+Metrics are sent to Data Prepper using the gRPC-based OpenTelemetry protocol.
+Data Prepper receives the metrics, dissects and maps their data points and saves each data point as an individual OpenSearch document.
+The diagram outlines the basic architecture for metrics ingestion using OpenTelemetry collector, Data Prepper and OpenSearch.
+
+![Architecture]({{ site.baseurl }}/assets/media/blog-images/2022-05-16-opentelemetry-metrics-for-data-prepper/architecture.drawio.png){: .img-fluid}
+
+The OpenTelemetry metrics data model is described in great detail in its [specification.](https://opentelemetry.io/docs/reference/specification/metrics/datamodel/)
+Very briefly, metrics are divided into 5 different types:
+
+* Sum
+* Gauge
+* Histogram
+* ExponentialHistogram
+* Summary
+
+Each metric type has a specific data point format.
+For example, sums and gauges use single value data points, while histograms and summaries used nested arrays to represent buckets of data.
+
+Correlation between different metrics as well as traces and logs is provided by attributes and trace/span ids.
+Direct connections to spans can be included with the Exemplars feature of OpenTelemetry.
+These exemplars will be mapped to an array in the generated document.
+
+The Data Prepper OpenTelemetry metrics support consists of two plugins: 
+
+* The OTel Metrics source to deserialize the gRPC data,
+* The OTel Metrics processor to preprocess the data points with regard to the metrics type.
+
+The following diagram shows a prototypical pipeline:
+
+![Pipeline]({{ site.baseurl }}/assets/media/blog-images/2022-05-16-opentelemetry-metrics-for-data-prepper/pipeline.drawio.png){: .img-fluid}
+
+Support for ExponentialHistograms is not yet included in Data Prepper 1.4 but will follow in a future version. 
+
+## Configuration
+
+Setting up the OpenTelemetry metrics support is straight-forward.
+You just need to configure the source and processor plugins: 
+
+```yaml
+metrics-pipeline:
+  source:
+    otel_trace_source:
+  processor:
+    - otel_metrics_raw_processor:
+  sink:
+    - opensearch:
+      hosts: [ "https://opensearch.local:9200" ]
+      username: username
+      password: password
+```
+
+The metric source supports the same configuration as the OpenTelemetry trace source.
+The metrics raw processor supports some properties for histograms:
+
+```yaml
+  processor:
+    - otel_metrics_raw_processor:
+      calculate_histogram_buckets: true
```

Comment:
At this point it seems we will be unable to get the exponential histogram into 1.4 since the build is still failing. These three configurations are only available with that PR.

Reply:
I will remove the documentation for those features than.